### PR TITLE
Add FilterInputType customization methods.

### PR DESCRIPTION
**Pull request by PascalSenn**
Extends the `FilterInputTypeDescriptor` by following methods: 

- `Name(NameString value)`
- `Name( Func<INamedType, NameString> createName)`
- `Description(value)`
- `Directive<TDirective>(TDirective directiveInstance)`
- `Directive<TDirective>()  where TDirective : class, new()`
- `Directive(NameString name,  params ArgumentNode[] arguments)`

Extends the `SortInputTypeDescriptor` by following methods: 

- `Name(NameString value)`
- `Name( Func<INamedType, NameString> createName)`
- `Description(value)`
- `Directive<TDirective>(TDirective directiveInstance)`
- `Directive<TDirective>()  where TDirective : class, new()`
- `Directive(NameString name,  params ArgumentNode[] arguments)`

Adresses #1149

### CODE REVIEW

File: src/Core/Types/Types/Descriptors/Helpers/DirectiveUtils.cs [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
+using System;
 using System.Collections.Generic;
 using System.Linq;
 using HotChocolate.Language;
 using HotChocolate.Types.Descriptors.Definitions;
 
 namespace HotChocolate.Types.Descriptors
 {
-    internal static class DirectiveUtils
+    public static class DirectiveUtils
```
Comment (PascalSenn) [inline src/Core/Types/Types/Descriptors/Helpers/DirectiveUtils.cs]: @michaelstaib  I had to open up this implementation for directive support

File: src/Core/Types/Types/Descriptors/Helpers/TypeNameHelper.cs [hunk 2]
```diff
@@ -5,7 +5,7 @@
 
 namespace HotChocolate.Types.Descriptors
 {
-    internal static class TypeNameHelper
+    public static class TypeNameHelper
```
Comment (PascalSenn) [inline src/Core/Types/Types/Descriptors/Helpers/TypeNameHelper.cs]: @michaelstaib  I had to open up this implementation for dynamic naming support

**Comment by michaelstaib** [comment]
There are still some flacky tests, I will  re-run the linux tests.

**Comment by michaelstaib** [comment]
The build does not provide correct coverage data :(

**Pull request merged**